sdk/nodejs/yaml/yaml.ts

```diff
@@ -2484,19 +2480,20 @@ import * as outputs from "../types/output";
     ):  pulumi.Output<{[key: string]: pulumi.CustomResource}> {
         const objs: pulumi.Output<{name: string, resource: pulumi.CustomResource}>[] = [];
 
-        for (const obj of config.objs) {
-            const fileObjects: pulumi.Output<{name: string, resource: pulumi.CustomResource}>[] =
-                parseYamlObject(obj, config.transformations, config.resourcePrefix, opts);
-            for (const fileObject of fileObjects) {
-                objs.push(fileObject);
+        pulumi.all([config.objs]).apply(([configObjs]) => {
+            for (const obj of configObjs) {
+                const fileObjects: pulumi.Output<{name: string, resource: pulumi.CustomResource}>[] =
+                    parseYamlObject(obj, config.transformations, config.resourcePrefix, opts);
+                for (const fileObject of fileObjects) {
+                    objs.push(fileObject);
+                }
             }
-        }
+        });
```

Comment:
I don't think this will work.  You are asynchronously running some code which mutates an array - then reading from that array before waiting on the asynchronous work to complete.

Reply:
I think this should be fixed now.